### PR TITLE
Fix compatibility with amphp/http-client beta

### PR DESCRIPTION
**Pull request by iggyvolz**
Fixes compatibility with the newest amphp/http-client release by updating revolt/event-loop to ^1.  Also sets the minimum stability to beta as all upstream dependencies have reached beta stage.

### CODE REVIEW

**Comment by iggyvolz** [comment]
Upon testing out more, it looks like this needs more compatibility fixes - will work on those in this branch

File: composer.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
             "email": "zoohie@gmail.com"
         }
     ],
-    "minimum-stability": "dev",
+    "minimum-stability": "beta",
```
Comment (iggyvolz) [inline composer.json]: Alternatively we could set minimum-stability stable and just pin to amphp/http-client v5.0.0-beta.11 (this strategy won't work for transitive beta/dev dependencies but all of amphp/http-client's dependencies are stable now)

File: test/Internal/PsrStreamBodyTest.php [hunk 2]
```diff
@@ -16,7 +16,6 @@ class PsrStreamBodyTest extends TestCase
      * @param int|null $size
      * @param int      $expectedSize
      *
-     * @return \Generator
```
Comment (iggyvolz) [inline test/Internal/PsrStreamBodyTest.php:19]: This line looks like a leftover from v1 - the method returns void.  Unrelated to the compatibility changes - can split this into a separate PR if desired

**Review by iggyvolz — COMMENTED** [review]
A couple quick notes/questions:

File: src/Internal/PsrStreamBody.php [hunk 3]
```diff
@@ -18,18 +18,18 @@ public function __construct(StreamInterface $stream)
...
-    public function createBodyStream(): ReadableStream
+    public function getContent(): ReadableStream
     {
         return new PsrInputStream($this->stream);
     }
 
-    public function getBodyLength(): ?int
+    public function getContentLength(): ?int
     {
         return $this->stream->getSize() ?? -1;
```
Comment (kelunik) [inline src/Internal/PsrStreamBody.php]: Fallback needs to be null here now.

File: test/Internal/PsrStreamBodyTest.php (hunk 2 above)
Comment (kelunik) [reply test/Internal/PsrStreamBodyTest.php:19]: It's fine here.

File: composer.json (hunk 1 above)
Comment (kelunik) [reply composer.json]: Removing and adding -beta to http client sounds good to me.

File: composer.json [hunk 4]
```diff
@@ -21,17 +21,15 @@
...
         }
     ],
-    "minimum-stability": "dev",
-    "prefer-stable": true,
     "require": {
         "php": ">=8.1",
         "amphp/amp": "^3",
         "amphp/byte-stream": "^2",
-        "amphp/http-client": "^5",
+        "amphp/http-client": "^v5.0.0-beta.11",
```
Comment (iggyvolz) [inline composer.json:28]: Should this be `^v5.0.0-beta.11` or `v5.0.0-beta.11`?  I don't think there's a difference but I'm not positive.

**Comment by trowski** [comment]
Superseded by #10. Thanks for your work!
